### PR TITLE
set session.new attribute

### PR DESCRIPTION
**Pull request by 0x155**
https://github.com/pallets/flask/issues/2564

### CODE REVIEW

File: flask/sessions.py [hunk 1]
```diff
@@ -61,21 +61,30 @@ def __init__(self, initial=None):
         def on_update(self):
             self.modified = True
             self.accessed = True
+            self.new = True
```
Comment (davidism) [inline flask/sessions.py]: Updating the session does not make it new.

File: flask/sessions.py [hunk 2]
```diff
@@ -61,21 +61,30 @@ def __init__(self, initial=None):
...
 
         super(SecureCookieSession, self).__init__(initial, on_update)
         self.modified = False
         self.accessed = False
+        if not hasattr(self, 'has_run'):
```
Comment (davidism) [inline flask/sessions.py]: This will always be true.

File: flask/sessions.py [hunk 3]
```diff
@@ -61,21 +61,30 @@ def __init__(self, initial=None):
...
+            self.has_run = True
+            self.new = False
+        else:
+            self.new = True
 
     def __getitem__(self, key):
         self.accessed = True
+        self.new = False
```
Comment (davidism) [inline flask/sessions.py]: Accessing data does not make the session not new.

**Comment by davidism** [comment]
Thanks for starting this, however I think you've misunderstood the `new` attribute. It should be `True` when an existing session is not loaded, it isn't affected by setting or accessing values.

**Comment by 0x155** [comment]
@davidism Could you tell me more about it? Yes I misunderstood the new attribute but I want to fix my mistake.

**Comment by davidism** [comment]
Either this is `self.new = initial is not None` or set in`SecureCookieSessionInterface.open_session`.

@mitsuhiko Is there a reason that this wasn't implemented that I'm missing?

**Comment by mitsuhiko** [comment]
@davidism I think that might have been just not well thought through.

**Comment by davidism** [comment]
Closing in favor of #2579.

**Pull request closed without merging**